### PR TITLE
Create ADR for Query Management V2

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes *Add ticket reference here*

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Changelog entry
~~- [ ] Attached issue to pull request~~
~~- [ ] Code is clear and sufficiently documented~~
~~- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)~~
~~- [ ] Correct, in line with design~~
~~- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

### CODE REVIEW

File: docs/query-managment-v2.md
Comment (LukasStordeur) [inline docs/query-managment-v2.md:1]: Can you make a folder for ADR's in general?

File: docs/query-managment-v2.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+1. Has the most features available
+2. Robust error handling
+3. Auto Garbage Collection
+4. Stale Time Configuration
+5. Handy DevTools
+6. More advanced mutation manipulation/configuration
+
+#### Cons
+
+1. some featuers have steeper learning curve
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:43]:
```suggestion
1. some features have a steeper learning curve
```

File: docs/query-managment-v2.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+## More Information
+
+[Feature-comparison between all mentioned libraries](https://tanstack.com/query/v4/docs/framework/react/comparison)
+[Comparison between SWR and React Query](https://dev.to/sakethkowtha/react-query-vs-useswr-122b)
+[Another comparison between SWR and React Query](https://www.dhiwise.com/post/data-on-demand-a-smackdown-of-swr-vs-react-query)
+[Comparison between RTK Query and React Query](https://www.frontendmag.com/insights/react-query-vs-rtk-query/#Pros_of_RTK_Query)
+
+# Use Plain JUnit5 for advanced test assertions
+
+## Context and Problem Statement
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:102]: From here on, I feel like you repeated some previous content. Is this on purpose?

File: docs/query-managment-v2.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+**Personally I advocate for React query, as more opinionated would mean easier, simpler and faster adoption into our codebase, with less code that has to be managed and maintained by us, which theoretically is the goal of moving from our own Query Management, the same goes for features, but that require conversation, whether SWR doesn't have all what we need.**
+
+## More Information
+
+[Feature-comparison between all mentioned libraries](https://tanstack.com/query/v4/docs/framework/react/comparison)
+[Comparison between SWR and React Query](https://dev.to/sakethkowtha/react-query-vs-useswr-122b)
+[Another comparison between SWR and React Query](https://www.dhiwise.com/post/data-on-demand-a-smackdown-of-swr-vs-react-query)
+[Comparison between RTK Query and React Query](https://www.frontendmag.com/insights/react-query-vs-rtk-query/#Pros_of_RTK_Query)
+
+# Use Plain JUnit5 for advanced test assertions
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:100]: I don't think this is part of the ADR?

File: docs/query-managment-v2.md
Comment (LukasStordeur) [inline docs/query-managment-v2.md:1]: Can you also add a file containing the template for future usages?

File: docs/query-managment-v2.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
+# Use React-Query as main Query Management tool, which idea to move away from our own implementation
+
+## Context and Problem Statement
+
+Currently we manage all of the queries to the backend ourself, which is burdened with a lot of code scattered all around the place.
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:5]:
```suggestion
Currently, we manage all of the queries to the backend ourselves, which is burdened with a lot of code scattered all around the place.
```

File: docs/query-managment-v2.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+This code is hard to read, require high effort to learn the flow, possible bugs or unwanted behavior is hard to track, finally adding new queries require a lot of boilerplate code.
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:6]:
```suggestion
The code is hard to read, requires high effort to learn the flow, possible bugs or unwanted behavior is hard to track, finally adding new queries requires a lot of boilerplate code.
```

File: docs/query-managment-v2.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Cons
+
+1. Requires a more boilerplate than competitors
+2. Bad, because increases bundle Size - [source](https://redux-toolkit.js.org/rtk-query/comparison#bundle-size)
+3. Bad, because has steep learning curve for developers new to Redux
+
+### Overview
+
+Out of most popular libraries that handles communication we have 3 possible contenders:
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:81]:
```suggestion
Out of most popular libraries that handle communication we have 3 possible contenders:
```

File: docs/query-managment-v2.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+- React Query
+- RTK Quert
+- SWR
+
+RTK Query can be filtered-out early, it heavily base itself on the Redux, which we don't use in the application, and we wouldn't like to, as it's unnecessary from the use-case of our application.
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:87]:
```suggestion
RTK Query can be filtered out early, it heavily bases itself on the Redux, which we don't use in the application, and we wouldn't like to, as it's unnecessary from the use-case of our application.
```

File: docs/query-managment-v2.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+The remaining two, are both lightweight, and simple, RQ more opinionated, more feature-rich(out-of-the-box), SWR gives more control in what we can do and in what way, but with price, of less features, and requires more time and effort into creating own implementation(manual caching, simpler error handling compared to react-query, no stale time configuration)
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:89]:
```suggestion
The remaining two, are both lightweight, and simple, RQ more opinionated, more feature-rich(out-of-the-box), SWR gives more control in what we can do and in what way, but with price, and fewer features, and requires more time and effort to create our own implementation(manual caching, simpler error handling compared to react-query, no stale time configuration)
```

With price, do you mean it has a paid license?

File: docs/query-managment-v2.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+**Personally I advocate for React query, as more opinionated would mean easier, simpler and faster adoption into our codebase, with less code that has to be managed and maintained by us, which theoretically is the goal of moving from our own Query Management, the same goes for features, but that require conversation, whether SWR doesn't have all what we need.**
```
Comment (LukasStordeur) [inline docs/query-managment-v2.md:91]:
```suggestion
**Personally I advocate for React query, as more opinionated would mean easier, simpler and faster adoption into our codebase, with less code that has to be managed and maintained by us, which theoretically is the goal of moving from our own Query Management, the same goes for features, but that requires conversation, whether SWR doesn't have all that we need.**
```

File: docs/query-managment-v2.md (hunk 8 above)
Comment (matborowczyk) [reply docs/query-managment-v2.md:89]: I think my brain jumped between thoughts I meant:
`but with price of fewer features, which results with more time and effort spent to create our own implementation`

Comment (LukasStordeur) [reply docs/query-managment-v2.md:89]: Hmm, okay, how about we keep just : with fewer features?

Comment (matborowczyk) [reply docs/query-managment-v2.md]: makes sense, changed

File: docs/query-managment-v2.md
Comment (matborowczyk) [reply docs/query-managment-v2.md:1]: Done

Comment (matborowczyk) [reply docs/query-managment-v2.md:1]: Added

File: docs/query-managment-v2.md (hunk 2 above)
Comment (matborowczyk) [reply docs/query-managment-v2.md]: I somehow overlook this, probably pasted it as I review the text in grammarly

File: docs/query-managment-v2.md (hunk 3 above)
Comment (matborowczyk) [reply docs/query-managment-v2.md]: Right, removed

File: docs/query-managment-v2.md (hunk 7 above)
Comment (matborowczyk) [reply docs/query-managment-v2.md]: changed

File: docs/query-managment-v2.md (hunk 5 above)
Comment (matborowczyk) [reply docs/query-managment-v2.md]: changed

**Comment by LukasStordeur** [comment]
don't forget to change the naming of the file according to the convention set by Core :D

**Review by LukasStordeur — APPROVED** [review]

**Pull request closed without merging**